freqtrade/data/dataprovider.py

```diff
@@ -28,18 +29,33 @@
 
 class DataProvider:
 
-    def __init__(self, config: dict, exchange: Optional[Exchange], pairlists=None) -> None:
+    def __init__(
+        self,
+        config: dict,
+        exchange: Optional[Exchange],
+        pairlists=None
+    ) -> None:
         self._config = config
         self._exchange = exchange
         self._pairlists = pairlists
         self.__cached_pairs: Dict[PairWithTimeframe, Tuple[DataFrame, datetime]] = {}
         self.__slice_index: Optional[int] = None
         self.__cached_pairs_backtesting: Dict[PairWithTimeframe, DataFrame] = {}
+        self.__external_pairs_df: Dict[PairWithTimeframe, Tuple[DataFrame, datetime]] = {}
```

Comment:
Having this track per producer instead of just per pair would enable multiple leaders. Something like:
```
self.__external_producer_pairs_df[producer_key][pair_key]
# or
self.__external_producer_pairs_df[producer_pair_key]
```